File: modules/flutter_channel/android/src/main/java/com/example/flutter_channel/Messages.java
package com.example.flutter_channel;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

class CommonParams {

    String key;
    Map<String, Object> arguments;

    public CommonParams(String key) {
        this.key = key;
    }

    public String getKey() {
        return key;
    }

    public Map<String, Object> getArguments() {
        return arguments;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public void setArguments(Map<String, Object> arguments) {
        this.arguments = arguments;
    }

    public Map<String, Object> toMap() {
        HashMap<String, Object> map = new HashMap<String, Object>();
        map.put("key", key);
        if (arguments != null) {
            map.put("arguments", arguments);
        }
        return map;
    }

}

class CallMethodParams {
    String methodName;
    Map<String, Object> arguments;

    public String getMethodName() {
        return methodName;
    }

    public void setMethodName(String methodName) {
        this.methodName = methodName;
    }

    public Map<String, Object> getArguments() {
        return arguments;
    }

    public void setArguments(Map<String, Object> arguments) {
        this.arguments = arguments;
    }

    public CallMethodParams(String methodName) {
        this.methodName = methodName;
    }

    @SuppressWarnings("unchecked")
    public static CallMethodParams fromMap(Map<String, Object> map) {
        String methodName = Objects.requireNonNull(map.get("methodName")).toString();
        CallMethodParams params = new CallMethodParams(methodName);
        params.arguments = (Map<String, Object>)map.get("arguments");
        return params;
    }

    public Map<String, Object> toMap() {
        HashMap<String, Object> map = new HashMap<String, Object>();
        map.put("methodName", methodName);
        if (arguments != null) {
            map.put("arguments", arguments);
        }
        return map;
    }
}

class ResponseParams {
    String methodName;
    Map<String, Object> arguments;

    public String getMethodName() {
        return methodName;
    }

    public void setMethodName(String methodName) {
        this.methodName = methodName;
    }

    public Map<String, Object> getArguments() {
        return arguments;
    }

    public void setArguments(Map<String, Object> arguments) {
        this.arguments = arguments;
    }

    public ResponseParams(String methodName) {
        this.methodName = methodName;
    }

    public Map<String, Object> toMap() {
        HashMap<String, Object> map = new HashMap<String, Object>();
        map.put("methodName", methodName);
        if (arguments != null) {
            map.put("arguments", arguments);
        }
        return map;
    }
}

File: modules/flutter_channel/ios/Classes/Messages.m
//
//  Messages.m
//  flutter_channel
//
//  Created by 赵光飞 on 2022/1/27.
//

#import "Messages.h"

@interface CommonParams ()

+ (CommonParams *)fromMap:(NSDictionary*)dict;
- (NSDictionary *)toMap;

@end

@implementation CommonParams

+ (CommonParams*)fromMap:(NSDictionary *)dict {
    CommonParams* result = [[CommonParams alloc] init];

  result.arguments = dict[@"arguments"];
  if ((NSNull *)result.arguments == [NSNull null]) {
    result.arguments = nil;
  }
  result.key = dict[@"key"];
  if ((NSNull *)result.key == [NSNull null]) {
    result.key = nil;
  }
  return result;
}

- (NSDictionary*)toMap {
    return [NSDictionary dictionaryWithObjectsAndKeys:(self.arguments ? self.arguments : [NSNull null]), @"arguments", (self.key ? self.key : [NSNull null]), @"key", nil];
}

@end


@interface CallMethodParams ()

+ (CallMethodParams *)fromMap:(NSDictionary*)dict;
- (NSDictionary *)toMap;

@end

@implementation CallMethodParams

+ (CallMethodParams*)fromMap:(NSDictionary *)dict {
    CallMethodParams* result = [[CallMethodParams alloc] init];

  result.arguments = dict[@"arguments"];
  if ((NSNull *)result.arguments == [NSNull null]) {
    result.arguments = nil;
  }
  result.methodName = dict[@"methodName"];
  if ((NSNull *)result.methodName == [NSNull null]) {
    result.methodName = nil;
  }
  return result;
}

- (NSDictionary*)toMap {
    return [NSDictionary dictionaryWithObjectsAndKeys:(self.arguments ? self.arguments : [NSNull null]), @"arguments", (self.methodName ? self.methodName : [NSNull null]), @"methodName", nil];
}

@end


@interface ResponseParams ()

+ (ResponseParams *)fromMap:(NSDictionary*)dict;
- (NSDictionary *)toMap;

@end

@implementation ResponseParams

+ (ResponseParams*)fromMap:(NSDictionary *)dict {
    ResponseParams* result = [[ResponseParams alloc] init];

  result.arguments = dict[@"arguments"];
  if ((NSNull *)result.arguments == [NSNull null]) {
    result.arguments = nil;
  }
  result.methodName = dict[@"methodName"];
  if ((NSNull *)result.methodName == [NSNull null]) {
    result.methodName = nil;
  }
  return result;
}

- (NSDictionary*)toMap {
    return [NSDictionary dictionaryWithObjectsAndKeys:(self.arguments ? self.arguments : [NSNull null]), @"arguments", (self.methodName ? self.methodName : [NSNull null]), @"methodName", nil];
}

@end
